sdks/java/extensions/sql/zetasql/src/test/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLDialectSpecTest.java

```diff
@@ -2457,6 +2458,256 @@ public void testDateFromUnixInt64() {
     pipeline.run().waitUntilFinish(Duration.standardMinutes(PIPELINE_EXECUTION_WAITTIME_MINUTES));
   }
 
+  /////////////////////////////////////////////////////////////////////////////
+  // TIME type tests
+  /////////////////////////////////////////////////////////////////////////////
+
+  @Test
+  public void testTimeLiteral() {
+    String sql = "SELECT TIME '15:30:00'";
```

Comment:
Can we test on a time having sub-second components? I wonder if `TIME` literal with microsecond precision works, because it needs to go through the "ZetaSQL -> Calcite -> Beam" code path.

Reply:
Done. TIME literal test with microsecond precision added ans passed.

Reply:
Tests with sub-second components added, including:
 - MILLISECOND
 - MICROSECOND

NANOSECOND precision are not supported for TIME type at this point.

Reply:
Nanosecond not supported is expected. The ZetaSQL parser itself is not supporting it by default.